packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -169,42 +176,49 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
   return getGradleVersionFor(androidPluginVersion ?? 'unknown');
 }
 
+/// Returns the gradle file from the top level directory.
+/// The returned file is not guaranteed to be present.
+File getGradleWrapperFile(Directory directory) {
+  return directory.childDirectory(gradleDirectoryName)
+      .childDirectory(gradleWrapperDirectoryName)
+      .childFile(gradleWrapperPropertiesFilename);
+}
+
+/// Parses the gradle wrapper distribution url to return a string containing
+/// the version number.
+///
+/// Expected input is of the form '...gradle-7.4.2-all.zip', and the output
```

Comment:
Can you add a test in gradle_utils_test for gradle-7.4.2-bin.zip and for rc candidates. That way refactors don't cause regressions assuming that users use the all resource. 
see https://services.gradle.org/distributions/ for example files.

Reply:
Added a test and also changed the gradle.org specific url to handle matching in the case of the rc distribution type.